pkg/noderesourcetopology/cache/overallocating.go

```diff
@@ -74,6 +74,9 @@ func NewOverallocating(lister listerv1alpha1.NodeResourceTopologyLister, indexer
 
 func (ov *Overallocating) GetByNode(nodeName string, pod *corev1.Pod) (*topologyv1alpha1.NodeResourceTopology, bool) {
 	logKey := keyFromPod(pod)
+
+	ov.resyncState(logKey)
```

Comment:
the commit message has enough context, but the explanation is poor and should be rephrased.
I spent some more time thinking about how to solve this issue, and I want to give a honest try to the "use an intermediate object" approach before to resort to the solution proposed here.
I think there is room for a clean and minimally invasive fix, which I'll experiment shortly.